src/main/java/com/crud/tasks/trello/client/TrelloClient.java

```diff
@@ -0,0 +1,54 @@
+package com.crud.tasks.trello.client;
+
+import com.crud.tasks.domain.TrelloBoardDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class TrelloClient {
+
+    @Value("${trello.api.endpoint.prod}")
+    private String trelloApiEndpoint;
+
+    @Value("${trello.app.key}")
+    private String trelloAppKey;
+
+    @Value("${trello.app.token}")
+    private String trelloToken;
+
+    @Value("${trello.app.username}")
+    private String username;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    public List<Optional<TrelloBoardDto>> getTrelloBoards() {
+        TrelloBoardDto[] boardsResponse = restTemplate.getForObject(buildUrl(), TrelloBoardDto[].class);
+        List<Optional<TrelloBoardDto>> boardsList = Arrays.asList(boardsResponse).stream()
```

Comment:
jesli boardResponse zwroci Ci nulla, to 
z tego 
`Arrays.asList(boardsResponse)`
poleci NullPointerException

Reply:
```
package com.crud.tasks.trello.client;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.test.util.ReflectionTestUtils;
import org.springframework.web.client.RestTemplate;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.when;

@RunWith(MockitoJUnitRunner.class)
public class TrelloClientTest {

    @Mock
    private RestTemplate restTemplate;

    @InjectMocks
    private TrelloClient trelloClient;

    @Test(expected = NullPointerException.class)
    public void getTrelloBoards() throws Exception {

        //given
        ReflectionTestUtils.setField(trelloClient, "trelloApiEndpoint", "http://random.com");
        ReflectionTestUtils.setField(trelloClient, "trelloAppKey", "key");
        ReflectionTestUtils.setField(trelloClient, "trelloToken", "token");
        ReflectionTestUtils.setField(trelloClient, "username", "?");
        when(restTemplate.getForObject(any(), any())).thenReturn(null);

        //when / then
        trelloClient.getTrelloBoards();

    }
}

Reply:
Przetestuj to sam